src/lib/plugin_apis/fs.api

```diff
@@ -2483,4 +2545,151 @@ gboolean bd_fs_udf_check_uuid (const gchar *uuid, GError **error);
  */
 BDFSUdfInfo* bd_fs_udf_get_info (const gchar *device, GError **error);
 
+/**
+ * bd_fs_btrfs_mkfs:
+ * @device: the device to create a new btrfs fs on
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the creation (right now
+ *                                                 passed to the 'mkfs.btrfs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a new btrfs fs was successfully created on @device or not
+ *
+ * Tech category: %BD_FS_TECH_BTRFS-%BD_FS_TECH_MODE_MKFS
+ *
+ */
+gboolean bd_fs_btrfs_mkfs (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_btrfs_wipe:
+ * @device: the device to wipe a UDF signature from
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the UDF signature was successfully wiped from the @device or
+ *          not
+ *
+ * Tech category: %BD_FS_TECH_BTRFS-%BD_FS_TECH_MODE_WIPE
+ */
+gboolean bd_fs_btrfs_wipe (const gchar *device, GError **error);
+
+/**
+ * bd_fs_btrfs_check:
+ * @device: the device containing the file system to check
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the check (right now
+ *                                                 passed to the 'btrfsck' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the filesystem was successfully checked or not
+ *
+ * Tech category: %BD_FS_TECH_BTRFS-%BD_FS_TECH_MODE_CHECK
+ */
+gboolean bd_fs_btrfs_check (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_btrfs_repair:
+ * @device: the device containing the file system to repair
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the repair (right now
+ *                                                 passed to the 'btrfs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the filesystem was successfully checked and repaired or not
+ *
+ * Tech category: %BD_FS_TECH_BTRFS-%BD_FS_TECH_MODE_REPAIR
+ */
+gboolean bd_fs_btrfs_repair (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_btrfs_set_label:
+ * @mpoint: the mount point of the file system to resize
+ * @label: label to set
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the label of UDF file system on the @mpoint was
+ *          successfully set or not
+ *
+ * Note: This function is intended to be used for btrfs filesystem on a single device,
+ *       for more complicated setups use the btrfs plugin instead.
+ *
+ * Tech category: %BD_FS_TECH_BTRFS-%BD_FS_TECH_MODE_SET_LABEL
+ */
+gboolean bd_fs_btrfs_set_label (const gchar *mpoint, const gchar *label, GError **error);
+
+/**
+ * bd_fs_btrfs_check_label:
+ * @label: label to check
+ * @error: (out) (allow-none): place to store error
+ *
+ * Returns: whether @label is a valid label for the UDF file system or not
```

Comment:
UDF?

Reply:
Yes, I decided to rename btrfs to UDF, to make it even more confusing :-)